OpenSim/Common/Test/testAPDMDataReader.cpp

```diff
@@ -0,0 +1,95 @@
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2017 Stanford University and the Authors                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "OpenSim/Common/DataAdapter.h"
+#include "OpenSim/Common/APDMDataReader.h"
+#include "OpenSim/Common/STOFileAdapter.h"
+#include <OpenSim/Auxiliary/auxiliaryTestFunctions.h>
+
+
+using namespace OpenSim;
+
+
+int main() {
+
+    try {
+        APDMDataReaderSettings readerSettings;
+        std::vector<std::string> imu_names{ "torso", "pelvis" };
+        std::vector<std::string> names_in_experiment{ "Static", "Middle" };
+        // Programmatically add items to name mapping, write to xml
+        for (int index = 0; index < imu_names.size(); ++index) {
+            ExperimentalSensor  nextSensor(names_in_experiment[index], imu_names[index]);
+            readerSettings.append_ExperimentalSensors(nextSensor);
+        }
+        readerSettings.print("apdm_reader.xml");
+        // read xml we wrote into a new APDMDataReader to readTrial
+        APDMDataReaderSettings reconstructFromXML("apdm_reader.xml");
+        APDMDataReader reader(reconstructFromXML);
+        DataAdapter::OutputTables tables = reader.extendRead("imuData01csv.csv");
+        // Write tables to sto files
+        // Accelerations
+        const TimeSeriesTableVec3& accelTableTyped =
+            APDMDataReader::getLinearAccelerationsTable(tables);
+        STOFileAdapterVec3::write(accelTableTyped, "accelerations.sto");
+        const SimTK::RowVectorView_<SimTK::Vec3>& rvv = accelTableTyped.getRowAtIndex(0);
+        SimTK::Vec3 fromTable = accelTableTyped.getRowAtIndex(0)[0];
```

Comment:
Verifying the first element in the first row may not be an adequate test. Would the last element of each table be a stronger test?

Reply:
We're testing both first and last row of the first table, all tables are treated the same so I'm not sure the test gets much stronger if we test all values but I'm open to do it.

Reply:
For completeness, I would go ahead and add the last row test to all of the tables.